src/Optimiser.cpp

```diff
@@ -522,12 +522,13 @@ double Optimiser::move_nodes(vector<MutableVertexPartition*> partitions, vector<
   // (2) - The quality function should be exactly the same value after
   //       aggregating/collapsing the graph.
 
+  vector<bool> comm_added(partitions[0]->n_communities(), false);
```

Comment:
This is now defined outside of the loop while `comms` is defined within the loop. Wouldn't it make more sense to define them in the same scope, so that we also define `comms` at this point already?

Reply:
Changed in 761d75c01a956fb9fe5dd0cfd2c718ff9a707741.

I moved the `comms` initialization up to `comm_added` in the two cases where this appears and placed a `comms.clear()` at the start of the relevant loops (where the old initialization used to be).